### ISSUE TITLE
Azbot bot show command with --msbot fails

### ISSUE THREAD

**Issue by vishwacsena** [Microsoft/botbuilder-tools#720]
## Tool
Name: AZ BOT
Version: AZ CLI: 2.0.46; botservice: 0.4.1
OS: *os version*

## Describe the bug
AZ bot show command optionally accepts a --msbot argument that can spit out the ABS configuration in a format that can be consumed by msbot. 
az bot show --msbot fails .. 

```bash
> az bot show -n cafebottest -g testCafeBot --msbot
The behavior of this command has been altered by the following extension: botservice
list index out of range
Traceback (most recent call last):
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\Lib\site-packages\knack\cli.py", line 197, in invoke
    cmd_result = self.invocation.execute(args)
  File "C:\Users\trdai\AppData\Local\Temp\pip-install-6nyhv9qe\azure-cli-core\azure\cli\core\commands\__init__.py", line 369, in execute
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\Lib\site-packages\six.py", line 693, in reraise
    raise value
  File "C:\Users\trdai\AppData\Local\Temp\pip-install-6nyhv9qe\azure-cli-core\azure\cli\core\commands\__init__.py", line 343, in execute
  File "C:\Users\trdai\AppData\Local\Temp\pip-install-6nyhv9qe\azure-cli-core\azure\cli\core\commands\__init__.py", line 182, in __call__
  File "C:\Users\trdai\AppData\Local\Temp\pip-install-6nyhv9qe\azure-cli-core\azure\cli\core\__init__.py", line 436, in default_command_handler
  File "C:\Users\vkannan\.azure\cliextensions\botservice\azext_bot\custom.py", line 225, in get_bot
    return create_bot_json(cmd, client, resource_group_name, resource_name, raw_bot_properties=raw_bot_properties)
  File "C:\Users\vkannan\.azure\cliextensions\botservice\azext_bot\custom.py", line 115, in create_bot_json
    app_password = [item['value'] for item in app_settings if item['name'] == 'MicrosoftAppPassword'][0]
IndexError: list index out of range
```

## To Reproduce
1. Install the latest AZ CLI and botservice extension
2. az login
3. try az bot show with --msbot option for one of your bots

## Expected behavior
AZ bot show works with --msbot option.

[bug]

**Comment by carlosscastro** [comment]
We are targeting this one for a fix on 11/26 to be released 12/4

**Comment by stevengum** [comment]
@vishwacsena, @carlosscastro, is this supposed to retrieve the MSA App password as well? Or is it expected that the user will go to apps.dev.microsoft.com to get/generate a password?

The bug here stems from that `create_bot_json()` does not receive the app_password arg so it attempts to retrieve it from the Web App's settings. This raises a problem for v4 bots (haven't tested yet with v3, but it should work just fine) because the password doesn't exist there, it now exists in the .bot file.

If  `az bot show --msbot true` should get the password as well, then there we can code for several cases:

1. There exists a bot file with the same name as the bot
2. A singular bot file exists, the name does not match the bot
3. Multiple bot files exist, the name does not match the bot

**Comment by stevengum** [comment]
> We need to read the appsettings for bot file path, use the botFileSecret to decrypt it (if encrypted) using the bot Framework configuration library and pick the app Secret from that bot file.

Observing this, it seems like the most correct thing to do would be to implement the botframework-config library in Python.

@carlosscastro, I know you're busy but we should discuss this item tonight/tomorrow.

**Comment by stevengum** [comment]
@vishwacsena is there an "easy and viable" path for using `az bot show --msbot true` on a registration bot? 

Otherwise we have a couple of paths to evaluate:

- Have them do the OAuth path for https://apps.dev.microsoft.com and then attempt to create a new password for them on their behalf
  - IIRC, a MSA App can only have two AppPws, if attempting to create a third AppPw this will fail
- Fail fast and have them manually go to https://apps.dev.microsoft.com to get their MSA AppId and a new AppPw
  - In this case we could output a partial IEndpointService entry for their .bot file.

We could do the partial IEndpointService entry for the first path as well in the case of failure.

**Comment by carlosscastro** [comment]
@stevengum the original bug from Vishwac was not implemented in az, we added this method last week and it seemed to work well. 

The password part is a new requirement and I think we are a bit late, let's discuss today.

**Comment by carlosscastro** [comment]
In registration bots we retrieve the password at creation time. In all cases including registration, we could get the password from the web app settings which we can query from az. Tom queries web app settings during msbot clone as well.

**Comment by stevengum** [comment]
Moved to 4.3 per conversation with Scott.

**Comment by stevengum** [comment]
Closed via Azure/Azure-cli#8035.
